heads.go

```diff
@@ -56,7 +56,14 @@ func (hh *heads) write(store ds.Write, c cid.Cid, height uint64) error {
 }
 
 func (hh *heads) delete(store ds.Write, c cid.Cid) error {
-	return store.Delete(hh.key(c))
+	err := store.Delete(hh.key(c))
+	// The go-datastore API currently says Delete doesn't return
+	// ErrNotFound, but it used to say otherwise.  Leave this
+	// here to be safe.
```

Comment:
This is misleading. `heads.delete` is in no way trying to implement the go-datastore API.

Reply:
The `heads` type is internal and is doing internal things, it's not a go-datastore instance.

Reply:
Re-reading, I understand the comment now.